src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -50,25 +51,70 @@
 
 public class SubstrateDispatcher {
 
-    private static volatile boolean run = true;
+    private enum Step {
+        COMPILE(),
+        LINK(COMPILE),
+        PACKAGE(LINK),
+        INSTALL(PACKAGE),
```

Comment:
I see that you don't do anything with `INSTALL` yet, so I guess that's for a follow up PR?

Reply:
Actually, it can be done in this PR, because at the moment `run` on the android target is actually `install` and `run` combined. So it's indeed best to split that functionality now as well.